### ISSUE TITLE
logrotate cannot find user starting with a digit in su directive

### ISSUE THREAD

**Issue by duritong** [logrotate/logrotate#67]
I have a user that starts with a digit: e.g. 4th.user_foo and logrotate is not able to find this user: 

```
# logrotate -d -v -f /etc/logrotate.d/php_4th.user
reading config file /etc/logrotate.d/php_4th.user
error: /etc/logrotate.d/php_4th.user:10 unknown user 'th.user_foo'
error: found error in /var/www/vhosts/4th.user/logs/php_error_log , skipping
removing last 1 log configs

Handling 0 logs
```

Yes, logrotate really tries to find 'th.user_foo' and not '4th.user_foo'

The config file looks like:

```
/var/www/vhosts/4th.user/logs/php_error_log {
  daily
  dateext
  missingok
  rotate 7
  compress
  copytruncate
  notifempty
  create 640 4th.user_foo 4th.user
  su 4th.user_foo 4th.user
}
```

The user exists:

```
# getent passwd 4th.user_foo
4th.user_foo:x:2034:2033:4th.user_foo:/var/www/vhosts/4th.user:/sbin/nologin
```

Based on the error message I would say it's pretty clear that the starting digit is cut off when parsing for the user on the su directive.

**Comment by glensc** [comment]
i've been on system, where user login was `900`, whose uid was `500` :D

**Comment by duritong** [comment]
btw this is

```
# logrotate --version
logrotate 3.8.6
```

on a CentOS 7

**Comment by kdudka** [comment]
I believe this was fixed in 6c0dfc4a via #53.  Feel free to reopen if the problem still exists in the upstream git HEAD.

**Comment by duritong** [comment]
I can confirm that is fixed by backporting the patch of  6c0dfc4 to the current logrotate version on EL7. Now we need to get it into EL7...

**Comment by kdudka** [comment]
I have filed a bug to track this:
https://bugzilla.redhat.com/1387533

If you are a customer of Red Hat, please tell Product Support that you are interested in a fix for this bug.
